### ISSUE TITLE
Documentation for trackUse/trackChange/trackAllow is very unclear

### ISSUE THREAD

**Issue by ezyang** [ndmitchell/shake#342]
I was looking for a polymorphic version of want, and I accidentally tried to use trackUse (apparently, I'm supposed to use apply!)

**Comment by ndmitchell** [comment]
Yes, the track functions are still evolving, better docs is a must (and perhaps better semantics) - apply is what you are after.

**Comment by ezyang** [comment]
Similarly, I think the Development.Shake.Rule would benefit from more guidance on how the pre-canned file operators can be reimplemented for custom rules. Or maybe it should have polymorphic versions of the operators.

**Comment by ndmitchell** [comment]
When you say precanned operators, do you mean things like `%>`, or things like `traceUse`, or both? Agreed, Shake.Rule should probably have some simple file rules as a demo.

**Comment by ezyang** [comment]
Also things like "need".
